File: backend/users/category.js
const { Schema, model } = require("mongoose");

const categorySchema= new Schema({
    title:{type:String, required: true},
    description:{type:String,default:""},
    updatedOn:{type:Date},
    createdOn:{tupe:Date}

})

categorySchema.pre('save', async function (next) {
    this.updatedOn= new Date();
    this.createdOn=new Date();
    next();
});

categorySchema.pre(['updateOne', 'findOneAndUpdate'], function (next) {
    const update = this.getUpdate();
    delete update._id;
    this.updatedOn = new Date();
    next();
});

const CategoryModel = model('MovieCategory',categorySchema);

module.exports=CategoryModel

File: frontend/VideoStreamming/lib/models/category.dart
class Categories {
  final String title;
  final String description;
  final String id;
  final String updatedOn;
  final int v;

  Categories({
    required this.title,
    required this.description,
    required this.id,
    required this.updatedOn,
    required this.v,
  });

  factory Categories.fromJson(Map<String, dynamic> json) {
    return Categories(
      title: json['title'],
      description: json['description'],
      id: json['_id'],
      updatedOn: json['updatedOn'],
      v: json['__v'],
    );
  }
}
